routers/web/repo/setting.go

```diff
@@ -1177,8 +1228,27 @@ func DeployKeysPost(ctx *context.Context) {
 	ctx.Redirect(ctx.Repo.RepoLink + "/settings/keys")
 }
 
+func DeleteSecret(ctx *context.Context) {
+	id := ctx.FormInt64("id")
```

Comment:
Alternatively, we could also delete by name repo, and owner.

Reply:
I'm not sure, maybe we can provide batch deletion later.

Reply:
That wasn't what I meant:
I simply meant that deletion by id is certainly the easiest way to implement it, but we could also have implemented it by using the secret name, owner, and repo. That makes it at least for users easier to see that the correct thing has been implemented.
But it is such an unimportant detail that we can end this discussion now and keep the current approach.

Reply:
@delvh While I agree with you, usually the user will never see this request because it is just a POST from the overview page. So the user will not see the called url.